src/wallet/rpc/spend.cpp

```diff
@@ -1564,8 +1565,9 @@ RPCHelpMan sendall()
                     pwallet->LockCoin(txin.prevout);
                 }
             }
-
-            return FinishTransaction(pwallet, options, rawTx);
+            auto result = FinishTransaction(pwallet, options, rawTx);
+            result.pushKV("weight", tx_size.weight);
```

Comment:
Not sure about using `tx_size`, we would be returning an estimation of the final size of tx. Not the real/current tx weight.

What about getting the current tx weight by serializing the `CMutableTransaction` object that appears inside `FinishTransaction` and explain in the RPC return value description that the returned weight will not be the final one if the transaction is not complete.

Reply:
Interesting, I agree with this point. Returning the exact size of the transaction would make the RPC definition more robust.

Reply:
This commit (along with this^ suggestion) [b3ac117](https://github.com/bitcoin/bitcoin/pull/29523/commits/b3ac1179ff90fe261af4e6dc9c7af64e1518b435) can be a separate PR as well, doesn't seem necessary for this PR to be merged if I didn't miss anything.

Reply:
Yes it's not needed.
I removed this commit
This can come as a separate PR, the size output can also be added to the other funding RPC's?